server/src/main/java/org/elasticsearch/synonyms/SynonymsManagementAPIService.java

```diff
@@ -313,65 +352,71 @@ private static IndexRequest createSynonymRuleIndexRequest(String synonymsSetId,
                 builder.field(SYNONYMS_SET_FIELD, synonymsSetId);
                 builder.field(SYNONYM_RULE_ID_FIELD, synonymRule.id());
                 builder.field(SYNONYMS_FIELD, synonymRule.synonyms());
+                builder.field(OBJECT_TYPE_FIELD, SYNONYM_RULE_OBJECT_TYPE);
             }
             builder.endObject();
 
             return new IndexRequest(SYNONYMS_ALIAS_NAME).id(internalSynonymRuleId(synonymsSetId, synonymRule.id()))
                 .opType(DocWriteRequest.OpType.INDEX)
-                .source(builder)
-                .id(internalSynonymRuleId(synonymsSetId, synonymRule.id()));
+                .source(builder);
+        }
+    }
+
+    private static IndexRequest createSynonymSetIndexRequest(String synonymsSetId) throws IOException {
+        try (XContentBuilder builder = XContentFactory.jsonBuilder()) {
+            builder.startObject();
+            {
+                builder.field(SYNONYMS_SET_FIELD, synonymsSetId);
+                builder.field(OBJECT_TYPE_FIELD, SYNONYM_SET_OBJECT_TYPE);
+            }
+            builder.endObject();
+
+            return new IndexRequest(SYNONYMS_ALIAS_NAME).id(synonymsSetId).opType(DocWriteRequest.OpType.INDEX).source(builder);
         }
     }
 
     private <T> void checkSynonymSetExists(String synonymsSetId, ActionListener<T> listener) {
-        client.prepareSearch(SYNONYMS_ALIAS_NAME)
-            .setQuery(QueryBuilders.termQuery(SYNONYMS_SET_FIELD, synonymsSetId))
-            .setSize(1)
-            .setPreference(Preference.LOCAL.type())
+        // Search for the synonym set ID
+        client.prepareGet(SYNONYMS_ALIAS_NAME, synonymsSetId)
             .execute(new DelegatingIndexNotFoundActionListener<>(synonymsSetId, listener, (l, searchResponse) -> {
-                if (searchResponse.getHits().getTotalHits().value == 0) {
+                if (searchResponse.isExists() == false) {
```

Comment:
👍

Reply:
should we rename `searchResponse` to `getResponse`?

Reply:
Done, thanks for the catch!